### PR TITLE
Payment Changes

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- change the payment changes for update
- changed the put method to post method
- added new API to get the payment status

### CODE REVIEW

File: controller/payment_controller.py [hunk 1]
```diff
@@ -180,23 +183,29 @@ def update_payment_status(
         payment_status: str
         ):
         try:
-            client_secret = response.get("client_secret")
+            client_secret = response.get("object").get("client_secret")
```
Comment (tamsud) [inline controller/payment_controller.py:186]: What would happen if the object itself is not there?

File: controller/payment_controller.py [hunk 2]
```diff
@@ -180,23 +183,29 @@ def update_payment_status(
...
             payment = PaymentReconciliation.make_request(
                 method="GET",
                 endpoint=f"/fhir/PaymentReconciliation?.detail.0.type.coding.0.display={client_secret}"
             )
             if payment.status_code == 200:
                 payment_data = payment.json()
                 payment_id = payment_data.get("entry", [{}])[0].get("resource", {}).get("id")
+                patient_id = payment_data.get("entry", [{}])[0].get("resource", {}).get("detail", [{}])[0].get("request", {}) \
+                    .get("reference").split("/")[-1]
```
Comment (tamsud) [inline controller/payment_controller.py:195]: check the length before taking the values using index position

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/payment_controller.py (hunk 1 above)
Comment (MonishaSivanathan) [reply controller/payment_controller.py]: i have handled it @tamsud

File: controller/payment_controller.py (hunk 2 above)
Comment (MonishaSivanathan) [reply controller/payment_controller.py]: checked the index nd picked the values, change the logic accordingly

File: controller/payment_controller.py [hunk 3]
```diff
@@ -180,23 +183,31 @@ def update_payment_status(
...
             payment = PaymentReconciliation.make_request(
                 method="GET",
                 endpoint=f"/fhir/PaymentReconciliation?.detail.0.type.coding.0.display={client_secret}"
             )
-            if payment.status_code == 200:
+            if payment.status_code == 200:  
                 payment_data = payment.json()
                 payment_id = payment_data.get("entry", [{}])[0].get("resource", {}).get("id")
+                patient_reference = payment_data.get("entry", [{}])[0].get("resource", {}).get("detail", [{}])[0].get("request", {}) 
+                if len(patient_reference) > 0:
```
Comment (tamsud) [inline controller/payment_controller.py:195]: why are you checking the length of the dict?

File: controller/payment_controller.py [hunk 4]
```diff
@@ -180,23 +183,31 @@ def update_payment_status(
...
+                    patient_id = patient_reference.get("reference").split("/")[-1]
+                appointment_reference = payment_data.get("entry", [{}])[0].get("resource", {}).get("detail", [{}])[0].get("response", {}) 
+                if len(appointment_reference) >= 0:
```
Comment (tamsud) [inline controller/payment_controller.py:198]: why are you checking the length of the dict?

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/payment_controller.py (hunk 3 above)
Comment (MonishaSivanathan) [reply controller/payment_controller.py]: i have checked the splited @tamsud

**Review by tamsud — APPROVED** [review]

**Pull request merged**